scim2-sdk-common/src/main/java/com/unboundid/scim2/common/BaseScimResource.java

```diff
@@ -176,7 +178,15 @@ public Set<String> getSchemaUrns()
    */
   public void setSchemaUrns(final Collection<String> schemaUrns)
   {
-    this.schemaUrns = new HashSet<String>(schemaUrns);
+    this.schemaUrns = new HashSet<>(schemaUrns);
```

Comment:
I think using a `LinkedHashSet` is a great idea because it would give clients a reliable way to list certain schemas before others. For example, if a resource has an extension schema, it's understandable to want that field to always be printed second.

I agree that it's more ideal to perform that check in the constructor. I'll make this change.

Reply:
This ended up being a bit simpler since the existing code already would throw an NPE if `null` was provided, so `schemaUrns` always has a non-null value. It was simplest to make an explicit `Objects.requireNonNull()` call in the constructor.